src/main/kotlin/string/addition/caculator/application/OperandParser.kt

```diff
@@ -0,0 +1,32 @@
+package string.addition.caculator.application
+
+import string.addition.caculator.domain.Operand
+
+object OperandParser {
+    private const val DEFAULT_DELIMITER_PATTERN = "[,:]"
+    private const val CUSTOM_DELIMITER_PATTERN = "//(.)\\\\n(.*)"
+
+    fun parse(inputStr: String): List<Operand> {
+        val customDelimiterResult = Regex(CUSTOM_DELIMITER_PATTERN).find(inputStr)
+        return if (customDelimiterResult == null) {
+            parseByDelimiter(
+                operandsStr = inputStr,
+                delimiter = DEFAULT_DELIMITER_PATTERN.toRegex()
+            )
+        } else {
+            parseByDelimiter(
+                operandsStr = customDelimiterResult.groupValues[2],
+                delimiter = customDelimiterResult.groupValues[1].toRegex()
+            )
+        }
+    }
+
+    private fun parseByDelimiter(operandsStr: String, delimiter: Regex): List<Operand> {
+        return if (operandsStr.isEmpty()) {
+            listOf(Operand.zero)
+        } else {
+            operandsStr.split(delimiter)
+                .map { Operand(it) }
+        }
```

Comment:
```suggestion
        return when {
            operandsStr.isEmpty() -> listOf(Operand.zero)
            else -> operandsStr.split(delimiter).map { Operand(it) }
        }
```
개인적으로는 if ~ else 식보다 when 을 사용하는 게 인덴트 상으로 보기 좋더라구요. 
취향의 영역이니 한 번 비교해보시고 마음에 드는 방식을 사용해주시면 될 것 같습니다!